kie-wb-common-react/boxed-expression-editor/src/components/ExpressionContainer/ExpressionContainer.tsx

```diff
@@ -61,11 +61,17 @@ export const ExpressionContainer: ({ selectedExpression }: ExpressionContainerPr
 
   const executeClearAction = useCallback(() => {
     setLogicTypeSelected(false);
-    setSelectedExpression((previousSelectedExpression: ExpressionProps) => ({
-      name: previousSelectedExpression.name,
-      dataType: previousSelectedExpression.dataType,
-      logicType: LogicType.Undefined,
-    }));
+    setSelectedExpression((previousSelectedExpression: ExpressionProps) => {
+      const updatedExpression = {
+        name: previousSelectedExpression.name,
+        dataType: previousSelectedExpression.dataType,
+        logicType: LogicType.Undefined,
+      };
+      if (window.beeApi?.resetExpressionDefinition) {
+        window.beeApi.resetExpressionDefinition(updatedExpression);
+      }
```

Comment:
You can avoid checking for the existence of `resetExpressionDefinition` before calling it.
With optional chaining you could just write `window.beeApi?.resetExpressionDefinition?.(...)`